containers/iam-reana-sync/generate_email_list.py

```diff
@@ -0,0 +1,43 @@
+import requests, json, os
+
+client_id = os.environ['CLIENT_ID']
+client_secret  = os.environ['CLIENT_SECRET']  
+      
+print(client_id)
+token_resp = requests.post(
+    "https://iam-escape.cloud.cnaf.infn.it/token",
+    data={
+        "grant_type": "client_credentials",
+        "client_id": client_id,
+        "client_secret": client_secret,
+        "scope": "scim:read"
+    },
+    headers={"Content-Type": "application/x-www-form-urlencoded"},
+)
+
+token_json = token_resp.json()
+token = token_json['access_token']
+headers = {"Authorization": "Bearer %s" % token}
+list_url = "https://iam-escape.cloud.cnaf.infn.it/scim/Users"
+
+startIndex = 1
+results = []
+resp = requests.get(list_url, headers=headers, params={"startIndex": startIndex})
+data = resp.json()
+response = json.loads(resp.text)
```

Comment:
Lines 25 to 27 are duplicated inside the `while`.
Both `resp` and `data` are objects (dicts) that once called, there are no need of being "re-called" again, so these lines can be taken out from the while.

Reply:
Also, I would avoid using `data`, as it is also used on line 9.

Reply:
I amended the `data` nomenclature and changed it to `resp_json`. 
As concerns the duplicate inside the loop, if the lines are taken out of the loop, it runs three times but it always fetches the first 100 users, so I am afraid I cannot take it out of the loop, and it needs to be there before it as the `while` statement needs to know what `response` is. I agree it is not pretty.. but works ;)

Reply:
Have a look to this Elena, it should solve the problem
https://www.geeksforgeeks.org/response-iter_content-python-requests/